src/extension/android/adb.ts

```diff
@@ -19,61 +19,36 @@ export enum AndroidAPILevel {
     GINGERBREAD_MR1 = 10,
 }
 
+enum KeyEvents {
+    KEYCODE_BACK = 4,
+    KEYCODE_DPAD_UP = 19,
+    KEYCODE_DPAD_DOWN = 20,
+    KEYCODE_DPAD_CENTER = 23,
+    KEYCODE_MENU = 82,
+}
+
 export enum DeviceType {
     AndroidSdkEmulator, // These seem to have emulator-<port> ids
     Other,
 }
 
-const AndroidSDKEmulatorPattern = /^emulator-\d{1,5}$/;
-
 export interface IDevice {
     id: string;
     isOnline: boolean;
     type: DeviceType;
 }
 
-export interface IAdb {
-    getConnectedDevices(): Q.Promise<IDevice[]>;
-    launchApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void>;
-    stopApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void>;
-    getOnlineDevices(): Q.Promise<IDevice[]>;
-    switchDebugMode(projectRoot: string, packageName: string, enable: boolean, debugTarget?: string): Q.Promise<void>;
-    apiVersion(deviceId: string): Q.Promise<AndroidAPILevel>;
-    reverseAdb(deviceId: string, packagerPort: number): Q.Promise<void>;
-}
-
-export abstract class AdbEnhancements implements IAdb {
-    public abstract getConnectedDevices(): Q.Promise<IDevice[]>;
-    public abstract launchApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void>;
-    public abstract stopApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void>;
-    public abstract switchDebugMode(projectRoot: string, packageName: string, enable: boolean, debugTarget?: string): Q.Promise<void>;
-    public abstract apiVersion(deviceId: string): Q.Promise<AndroidAPILevel>;
-    public abstract reverseAdb(deviceId: string, packagerPort: number): Q.Promise<void>;
-
-    public getOnlineDevices(): Q.Promise<IDevice[]> {
-        return this.getConnectedDevices().then(devices => {
-            return devices.filter(device =>
-                device.isOnline);
-        });
-    }
-}
-
-export class Adb extends AdbEnhancements {
-    private childProcess: ChildProcess;
-    private commandExecutor: CommandExecutor;
+const AndroidSDKEmulatorPattern = /^emulator-\d{1,5}$/;
 
-    constructor({childProcess = new ChildProcess(), commandExecutor = new CommandExecutor()} = {}) {
-        super();
-        this.childProcess = childProcess;
-        this.commandExecutor = commandExecutor;
-    }
+export class AdbHelper {
```

Comment:
Nit: I'd suggest to keep `Adb` name to a) avoid redundant renaming and b) reflect that this is just an abstraction over adb

Reply:
I believe it isn't just `abstraction over adb`, here also implemented some helper functions.